pints/_mcmc/_monomial_gamma_hmc.py

```diff
@@ -0,0 +1,507 @@
+        self._current_energy = None     # Aka U(current_q) = -log_pdf
+        self._current_gradient = None
+        self._current_momentum = None   # Aka current_p
+
+        # Current point in the leapfrog iterations
+        self._momentum = None       # Aka p in the chapter
+        self._position = None       # Aka q in the chapter
+        self._gradient = None       # Aka grad_U(q) in the chapter
+
+        # Iterations, acceptance monitoring, and leapfrog iterations
+        self._mcmc_iteration = 0
+        self._mcmc_acceptance = 0
+        self._frog_iteration = 0
+
+        # Default number of leapfrog iterations
+        self._n_frog_iterations = 20
+
+        # Default integration step size for leapfrog algorithm
+        self._epsilon = 0.1
+        self._step_size = None
+        self.set_leapfrog_step_size(np.diag(self._sigma0))
+
+        # Divergence checking
+        # Create a vector of divergent iterations
+        self._divergent = np.asarray([], dtype='int')
+
+        # Default threshold for Hamiltonian divergences
+        # (currently set to match Stan)
+        self._hamiltonian_threshold = 10**3
+
+        # Estimate quantities needed for sampling from the ke distribution
+        self._a = 1.0
+        self._c = 0.2
+        self._m = 1.0
+        self._initialise_ke()
+
+    def a(self):
+        """
+        Returns `a` in kinetic energy function
+        """
+        return self._a
+
+    def c(self):
+        """
+        Returns `c` in kinetic energy function
+        """
+        return self._c
+
+    def m(self):
+        """
+        Returns `m` in kinetic energy function
+        """
+        return self._m
+
+    def set_a(self, a):
+        """
+        Sets `a` in kinetic energy function
+        """
+        if a <= 0:
+            raise ValueError("a must be positive")
+        self._a = a
+        self._initialise_ke()
+
+    def set_c(self, c):
+        """
+        Sets `c` in kinetic energy function
+        """
+        if c <= 0:
+            raise ValueError("c must be positive")
+        self._c = c
+        self._initialise_ke()
+
+    def set_m(self, m):
+        """
+        Sets mass `m` in kinetic energy function
+        """
+        if m <= 0:
+            raise ValueError("a must be positive")
+        self._m = m
+        self._initialise_ke()
+
+    def _initialise_ke(self):
+        """
+        Initialises functions needed for sampling from soft kinetic energy
+        function defined in [1]
+        """
+        self._z = integrate.quad(
+            lambda p: np.exp(-self._K_indiv(p, self._a, self._c, self._m)),
+            -float('Inf'), float('Inf'))[0]
+        self._f = self._inverse_cdf_calculator(self._a, self._c,
+                                               self._m, self._z)
+
+    def _g(self, p, a, m):
+        """
+        Helper function used to define soft kinetic energy in [1]
+        """
+        return (1.0 / m) * np.sign(p) * np.abs(p)**(1.0 / a)
```

Comment:
They can stay, but you don't need the `.0` bits

Reply:
Done